### PR TITLE
2.1 devel port fcoemon patches from rhel

### PR DESCRIPTION
**Pull request by rvykydal**
Rebasing https://github.com/storaged-project/blivet/pull/447
on 2.1-devel

### CODE REVIEW

File: blivet/fcoe.py [hunk 1]
```diff
@@ -120,27 +121,75 @@ def add_san(self, nic, dcb=False, auto_vlan=True):
...
+                time.sleep(1)
+
+            timeout_msg = "retrying to set up dcb for fcoe"
+            while timeout > 0:
+                rc, out = util.run_program_and_capture_output(
+                    ["dcbtool", "sc", nic, "app:fcoe", "e:1", "a:1", "w:1"])
+                if rc == 0:
+                    break
+                timeout -= 1
+                time.sleep(1)
```
Comment (vpodzime) [inline blivet/fcoe.py]: Looks like we could make use of a function for this waiting loop.

**Comment by rvykydal** [comment]
@vpodzime 
New version: instead of using a function for the loop I used a for loop iterating over commands.
I also noticed that formerly the timeout_msg was wrong in case of timeout (always the last one).

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
From the CI:

```
*** Running pylint ***
PYTHONPATH=.:tests/: tests/pylint/runpylint.py
************* Module /home/jenkins/workspace/blivet-2.x-PR/blivet/fcoe.py
W0631(undefined-loop-variable):158,42: FCoE.add_san: Using possibly undefined loop variable 'timeout_msg'
make[1]: *** [Makefile:76: pylint] Error 4
```

**Comment by rvykydal** [comment]
jenkins, test this please

**Comment by rvykydal** [comment]
I updated the PR with fix for pylint, the actual change is here:
https://github.com/rvykydal/blivet/commit/4a07c6659b7cc90cd345361440128709fa8a975b

**Pull request merged**